GoogleChromeLabs/ps-analysis-tool
# Feature: Detect use of Google Sign In (GSI) libraries

## Description

Detect use of Google Sign In (GSI) and GIS libraries and provide guidance to users. This PR covers the following

- Add a new section on the cookie's landing page specifically for known breakages.
- The PSAT tool should automatically detect GIS and GSI on any website.
- It should also parse and identify the code signatures, and list them for each library.
- The developer should be directed to the relevant documentation link for migration.
- The code structure should allow for easy addition of more libraries for detection.

## Relevant Technical Choices

- Create a new package named `@ps-analysis-tool/libray-detection`
- Allow new components to be added from `CookiesLanding` components in `@ps-analysis-tool/design-system`
- Add new section for known-breakages from the `@ps-analysis-tool/extension`  in `Cookies` component
- Create `core`, `libraries`, `components` and `utils` directories in the new library-detection package.
- Create configuration for adding more libraries in `config.ts`

## Testing Instructions
- Visit any website for example which may have GIS or GIS integration for example linkedin.com (logged out state)
- Go to cookies landing page and check if you able to see detected libraries under "Know Breakages" section.
- Similarly visit websites that do not have GSI feature or just one feature for example scripts.com


## Additional Information:

- The detection may only work for 80% websites. We would improve it in future versions.
- It is possible that the tool may also produce false positive results on certain websites. However, we have not encountered such a case so far.




## Screenshot/Screencast

<img width="782" alt="image" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/6297436/b8a12518-9918-46f2-ad7a-96668d20c001">


## Checklist






Fixes #316
`
